File: Monstone_Android_Client/app/src/main/java/com/example/nightwingky/monstone/main_program/me/change_user_info/ChangeUserInfoAdapter.java
package com.example.nightwingky.monstone.main_program.me.change_user_info;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.nightwingky.monstone.R;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by nightwingky on 17-3-13.
 */

public class ChangeUserInfoAdapter extends BaseAdapter {

    private LayoutInflater mInflater;

    private List<String> titleList;
    private List<Object> infoList;

    public ChangeUserInfoAdapter(Context context, ChangeUserInfoVO changeUserInfoVO) {
        this.mInflater = LayoutInflater.from(context);

        this.titleList = new ArrayList<String>();
        this.titleList.add("头像");
        this.titleList.add("昵称");
        this.titleList.add("性别");
        this.titleList.add("生日");
        this.titleList.add("手机");
        this.titleList.add("邮箱");
        this.titleList.add("简介");

        this.infoList = new ArrayList<Object>();
        this.infoList.add(changeUserInfoVO.getPortrait());
        this.infoList.add(changeUserInfoVO.getName());
        switch (changeUserInfoVO.getGender()) {
            case 0:
                this.infoList.add("男");
                break;
            case 1:
                this.infoList.add("女");
                break;
            default:
                this.infoList.add("保密");
                break;
        }
        this.infoList.add(changeUserInfoVO.getBirth());
        this.infoList.add(String.valueOf(changeUserInfoVO.getMobile()));
        this.infoList.add(changeUserInfoVO.getEmail());
        this.infoList.add(changeUserInfoVO.getIntroduction());
    }

    @Override
    public int getCount() {
        return infoList.size();
    }

    @Override
    public Object getItem(int position) {
        return infoList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (position == 0) {
            ViewHolderPortrait viewHolderPortrait= new ViewHolderPortrait();
            convertView = mInflater.inflate(R.layout.item_portrait_change_user_info, null);
            viewHolderPortrait.tv_portrait = (TextView) convertView.findViewById(R.id.tv_portrait_change_user_info);
            viewHolderPortrait.img_portrait = (ImageView) convertView.findViewById(R.id.img_portrait_change_user_info);

            convertView.setTag(viewHolderPortrait);

            viewHolderPortrait.img_portrait.setImageResource(R.mipmap.ic_launcher);

            return convertView;
        } else {
            ViewHolderElse viewHolderElse = new ViewHolderElse();
            convertView = mInflater.inflate(R.layout.item_else_change_user_info, null);
            viewHolderElse.tv_else = (TextView) convertView.findViewById(R.id.tv_else_change_user_info);
            viewHolderElse.tv_info = (TextView) convertView.findViewById(R.id.tv_info_else_change_user_info);

            convertView.setTag(viewHolderElse);

            viewHolderElse.tv_else.setText(titleList.get(position));
            viewHolderElse.tv_info.setText(String.valueOf(infoList.get(position)));

            if (position == 6) {
                viewHolderElse.tv_info.setText("生命中，有风，有雨，但别忘了也会有阳光");
            }

            return convertView;
        }

    }

    class ViewHolderPortrait {
        public TextView tv_portrait;
        public ImageView img_portrait;
    }

    class ViewHolderElse {
        public TextView tv_else;
        public TextView tv_info;
    }
}

File: kotlin_client/app/src/main/java/com/graduation/nightwingky/book_drifting/monstone/main_program/fragmentMe/change_user_info/ChangeUserInfoAdapter.kt
package com.graduation.nightwingky.book_drifting.monstone.main_program.fragmentMe.change_user_info

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView

import com.graduation.nightwingky.book_drifting.R

import java.util.ArrayList

/**
 * Created by nightwingky on 17-3-13.
 */

class ChangeUserInfoAdapter(context: Context, changeUserInfoVO: ChangeUserInfoVO) : BaseAdapter() {

    private val mInflater: LayoutInflater

    private val titleList: MutableList<String>
    private val infoList: MutableList<Any>

    init {
        this.mInflater = LayoutInflater.from(context)

        this.titleList = ArrayList()
        this.titleList.add("头像")
        this.titleList.add("昵称")
        this.titleList.add("性别")
        this.titleList.add("生日")
        this.titleList.add("手机")
        this.titleList.add("邮箱")
        this.titleList.add("简介")

        this.infoList = ArrayList()
        this.infoList.add(changeUserInfoVO.portrait!!)
        this.infoList.add(changeUserInfoVO.name!!)
        when (changeUserInfoVO.gender) {
            0 -> this.infoList.add("男")
            1 -> this.infoList.add("女")
            else -> this.infoList.add("保密")
        }
        this.infoList.add(changeUserInfoVO.birth!!)
        this.infoList.add(changeUserInfoVO.mobile.toString())
        this.infoList.add(changeUserInfoVO.email!!)
        this.infoList.add(changeUserInfoVO.introduction!!)
    }

    override fun getCount(): Int {
        return infoList.size
    }

    override fun getItem(position: Int): Any {
        return infoList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
        var convertView = convertView
        if (position == 0) {
            val viewHolderPortrait = ViewHolderPortrait()
            convertView = mInflater.inflate(R.layout.item_portrait_change_user_info, null)
            viewHolderPortrait.tv_portrait = convertView.findViewById<View>(R.id.tv_portrait_change_user_info) as TextView
            viewHolderPortrait.img_portrait = convertView.findViewById<View>(R.id.img_portrait_change_user_info) as ImageView

            convertView.tag = viewHolderPortrait

            viewHolderPortrait.img_portrait!!.setImageResource(R.drawable.portrait)

            return convertView
        } else {
            val viewHolderElse = ViewHolderElse()
            convertView = mInflater.inflate(R.layout.item_else_change_user_info, null)
            viewHolderElse.tv_else = convertView.findViewById<View>(R.id.tv_else_change_user_info) as TextView
            viewHolderElse.tv_info = convertView.findViewById<View>(R.id.tv_info_else_change_user_info) as TextView

            convertView.tag = viewHolderElse

            viewHolderElse.tv_else!!.text = titleList[position]
            viewHolderElse.tv_info!!.text = infoList[position].toString()

            if (position == 6) {
                viewHolderElse.tv_info!!.text = "生命中，有风，有雨，但别忘了也会有阳光"
            }

            return convertView
        }

    }

    internal inner class ViewHolderPortrait {
        var tv_portrait: TextView? = null
        var img_portrait: ImageView? = null
    }

    internal inner class ViewHolderElse {
        var tv_else: TextView? = null
        var tv_info: TextView? = null
    }
}
